src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Weekday_StronglyTypedEnum.txt

```diff
@@ -0,0 +1,103 @@
+
+// Sunday, Start on Thursday
+>> Weekday(Date(2016, 1, 3),14)
+Errors: Error 25-27: Invalid argument type (Decimal). Expecting a Enum (StartOfWeek) value instead.
+
+// Sunday, Start on Friday
+>> Weekday(Date(2016, 1, 3),15)
+Errors: Error 25-27: Invalid argument type (Decimal). Expecting a Enum (StartOfWeek) value instead.
+
+// Sunday, Start on Saturday
+>> Weekday(Date(2016, 1, 3),16)
+Errors: Error 25-27: Invalid argument type (Decimal). Expecting a Enum (StartOfWeek) value instead.
+
+// **************** INVALID EXCEL CODES AS STARTOFWEEK PARAMETER  ****************
+
+// 0 not in list of supported StartofWeek Excel codes
+>> Weekday(Date(2016, 1, 3),0)
+Errors: Error 25-26: Invalid argument type (Decimal). Expecting a Enum (StartOfWeek) value instead.
+
+// 5 not in list of supported StartofWeek Excel codes
+>> Weekday(Date(2016, 1, 3),5)
+Errors: Error 25-26: Invalid argument type (Decimal). Expecting a Enum (StartOfWeek) value instead.
+
+// Negative error code not supported
+>> Weekday(Date(2016, 1, 3),-12)
+Errors: Error 25-26: Invalid argument type (Decimal). Expecting a Enum (StartOfWeek) value instead.
+
+>> Weekday(DateValue("1/2/2022"),1E10)
+Errors: Error 30-34: Invalid argument type (Decimal). Expecting a Enum (StartOfWeek) value instead.
+
+// ******************************** STRING PARAMETERS ********************************
+
+// Valid StartOfWeek excel code passed as String
+>> Weekday(DateValue("1/2/2022"),"12")
+Errors: Error 30-34: Invalid argument type (Text). Expecting a Enum (StartOfWeek) value instead.
+
+// Invalid string passed as StartofWeek parameter
+>> Weekday(Date(2016, 1, 3),"abc")
+Errors: Error 25-30: Invalid argument type (Text). Expecting a Enum (StartOfWeek) value instead.
+
+// ******************************** BOOLEAN PARAMETERS ********************************
+
+// Boolean value true coerced to Excel code 1
+>> Weekday(DateValue("1/2/2022"),true)
+Errors: Error 30-34: Invalid argument type (Boolean). Expecting a Enum (StartOfWeek) value instead.
+
+// Boolean value false coerced to Excel code 0, which is unsupported StartOfWeek code
+>> Weekday(DateValue("1/2/2022"),false)
+Errors: Error 30-35: Invalid argument type (Boolean). Expecting a Enum (StartOfWeek) value instead.
+
+
+// ******************************** NULL AND ERROR PARAMETERS ********************************
+
+// Saturday, Empty string "" as StartOfWeek parameter
+>> Weekday(Date(2016, 1, 9),"")
+Errors: Error 25-27: Invalid argument type (Text). Expecting a Enum (StartOfWeek) value instead.
+
+// Sunday, Empty string "" as StartOfWeek parameter
+>> Weekday(Date(2022,1,2),"")
+Errors: Error 23-25: Invalid argument type (Text). Expecting a Enum (StartOfWeek) value instead.
+
+// Empty string as Date and StartOfWeek parameter
+>> Weekday("","")
+Errors: Error 11-13: Invalid argument type (Text). Expecting a Enum (StartOfWeek) value instead.
```

Comment:
also a test that passing another enum (not StartOfWeek) is erroneous?

Reply:
Added